### PR TITLE
Improved the lesson - Added more theory, examples, formatting

### LINKED ISSUE #63: improve:  Space Complexity
### ⭐ Summary

Please improve the documentation for the lesson space complexity, try to make to better.

If anyone wants to do it, I'll assign him.

### 💻 Location

section: 7
lesson: C
order: 7C
filename: [space-complexity.md](https://github.com/Utkarsh1504/DSA-Java/blob/main/lessons/space-complexity.md)

### ❌ Additional Info

You can take the help of other markdown files in the [/lesson](https://github.com/Utkarsh1504/DSA-Java/blob/main/lessons/loops.md) directory so that you get the idea of in which fashion you should write.

If you don't know how to do markdown, don't worry it is quite easy, just follow this docs-https://guides.github.com/features/mastering-markdown/

If your concepts are not that good follow this space and time complexity video of @kunal-kushwaha (https://www.youtube.com/watch?v=mV3wrLBbuuE)

### PR DESCRIPTION
**Pull request by ananya0504**
Fixes #63

#### Short description of what this resolves:

improve: Space Complexity #63

#### What Changes proposed in this pull request:
- Added more theory
- Removed the first code snippet and replaced it with a better and easily understandable one
- Added 1 code snippet for finding space complexity in arrays
- Improved overall formatting
- Improved the overall language

#### Checklist

- [x] Added description of change
- [x] Added code examples, the test must pass
- [x] Added documentation so that the program is self-explanatory and educational.
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/Utkarsh1504/DSA-Java/blob/main/CONTRIBUTING.md#requesting-a-lesson)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 
- Improvised on the language of the lesson.
- Removed certain lines which were difficult to understand for beginners
- Added necessary formatting

### CODE REVIEW

File: lessons/space-complexity.md [hunk 1]
```diff
@@ -6,52 +6,102 @@ section: "Space & Time Complexity"
...
+*Above all, it’s necessary to mention that space complexity depends on a variety of things such as the programming language, the compiler, or even the machine running the algorithm.*
+
+## Memory Usage while Execution
+While executing, an algorithm uses memory space for three reasons:
+- **Instruction Space**
+-- It's the amount of memory used to save the compiled version of instructions.
+
+- **Environmental Stack**
+-- Sometimes an algorithm(function) may be called inside another algorithm(function). In such a situation, the current variables are pushed onto the system stack, where they wait for further execution and then the call to the inside algorithm(function) is made.\
+Ex.  If a function A() calls function B() inside it, then all th variables of the function A() will get stored on the system stack temporarily, while the function B() is called and executed inside the funciton A().
```
Comment (Utkarsh1504) [inline lessons/space-complexity.md:24]:
```suggestion
Ex.  If a function A() calls function B() inside it, then all the variables of the function A() will get stored on the system stack temporarily, while the function B() is called and executed inside the function A().
```
please fix this one

**Review by Utkarsh1504 — CHANGES_REQUESTED** [review]
All looks great, you have improved it in a great way, just fix that typo, and I'll merge your PR.
Thank you!

**Comment by ananya0504** [comment]
Okay so I did fixed the typo, and I commit and push it onto my repo

File: lessons/space-complexity.md (hunk 1 above)
Comment (ananya0504) [reply lessons/space-complexity.md]: Done!

**Review by Utkarsh1504 — APPROVED** [review]
Awesome, Looks good.
Thank you!

**Pull request merged**